### PR TITLE
CLN/DOC: Refactor timeseries.rst intro and overview

### PR DESCRIPTION
**Pull request by mroeschke**
Refactoring `timeseries.rst` introduction and overview to give a better introduction into pandas timeseries functionality:

- Introduce a wider range of examples
- Overview all scalar and array classes, related data types, and range methods, and `NaT` in the beginning

### CODE REVIEW

**Comment by mroeschke** [comment]
screen shot 2018-09-15 at 10 56 12 pm
screen shot 2018-09-15 at 10 56 23 pm
screen shot 2018-09-15 at 10 56 32 pm
screen shot 2018-09-15 at 10 56 40 pm

File: doc/source/timeseries.rst [hunk 1]
```diff
@@ -21,69 +21,112 @@
...
-   # Daily means
-   ts.resample('D').mean()
+    ts = pd.Timestamp('2018-01-05')
+    ts.day_name()
+    # Add 1 day
+    saturday = ts + pd.Timedelta('1 day')
+    saturday.day_name()
+    # Add 1 business day (Friday --> Monday)
+    monday = ts + pd.tseries.offsets.BDay()
+    monday.day_name()
```
Comment (datapythonista) [inline doc/source/timeseries.rst]: may be we could rename `ts` to `friday`? I think it'd make the example more consistent and easier to understand.

File: doc/source/timeseries.rst [hunk 2]
```diff
@@ -21,69 +21,112 @@
...
-Change frequency and fill gaps:
+Resampling or converting a time series to a particular frequency
 
 .. ipython:: python
 
-   # to 45 minute frequency and forward fill
-   converted = ts.asfreq('45Min', method='pad')
-   converted.head()
+   idx = pd.date_range('2018-01-01', periods=72, freq='H')
+   ts = pd.Series(np.random.randn(len(idx)), index=idx)
```
Comment (datapythonista) [inline doc/source/timeseries.rst]: I'd prefer `range` over `randn`. The data won't also be deterministic, but when you do the mean, it'll be more clear, and the means will increase.

Also, may be it's difficult, but if you can find a resampling example where the original is much smaller than 72, I think it'd help users if we show the original data before showing the transformed data.

File: doc/source/timeseries.rst [hunk 3]
```diff
@@ -1443,7 +1486,7 @@ time. The method for this is :meth:`~Series.shift`, which is available on all of
 the pandas objects.
 
 .. ipython:: python
-
+   ts = pd.Series(np.random.randn(len(rng)), index=rng)
```
Comment (datapythonista) [inline doc/source/timeseries.rst]: same as before regarding `random` vs `range`

File: doc/source/timeseries.rst [hunk 4]
```diff
@@ -21,69 +21,112 @@
...
+   pd.Series(range(3), index=pd.date_range('2000', freq='D', periods=3))
+
+However, :class:`Series` and :class:`DataFrame` can directly also support the time component as data itself.
+
+.. ipython:: python
+
+   pd.Series(pd.date_range('2000', freq='D', periods=3))
+
+:class:`Series` and :class:`DataFrame` have extended data type support and functionality for ``datetime`` and ``timedelta``
+data when the time data is used as data itself. The other time related concepts will be stored as ``object`` data.
```
Comment (datapythonista) [inline doc/source/timeseries.rst]: I'd name "The other time related concepts", and may be show an example with a datetime Series, so it's easier to see what we exactly mean here.

**Review by datapythonista — COMMENTED** [review]
Looks great, just added some ideas that I think would make the doc clearer.

**Comment by mroeschke** [comment]
Good suggestions @datapythonista. Incorporated all your comments.

**Review by datapythonista — APPROVED** [review]
Great work, looks perfect.

**Review by WillAyd — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @datapythonista @WillAyd.

**Pull request merged**